File: ruby/easy/student_attendance_record_i.rb
# @param {String} s
# @return {Boolean}
def check_record(s)
    i, a_count, l_count = 0, 0, 0
    while i < s.length
        if s[i] == 'L'
            l_count += 1
            i += 1 and next
        elsif s[i] == 'A'
            a_count += 1
        end
        return false if l_count > 2 || a_count > 1
        l_count = 0
        i += 1
    end
    return l_count <= 2 && a_count <= 1
end

File: javascript/easy/studentAttendanceRecordI.js
/**
 * @param {string} s
 * @return {boolean}
 */
var checkRecord = function(s) {
    let aCount = 0, lCount = 0, i = 0;
    while(i < s.length) {
        if(s[i] === 'L') {
            lCount++;
            i++; continue;
        }
        if(s[i] === 'A') {
            aCount++;
        }
        
        if(lCount > 2 || aCount > 1) {
            return false;
        }
        lCount = 0;
        i++;
    }
    return (aCount <= 1) && (lCount <= 2);
};
